fix-block-img-tag-sizes.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+// Adds 'correct' w & h values to img tags in core/image blocks
+
+function fix_img_block_sizes($block_content, $block, $instance)
+{
+
+    // Use resize args for images instead of w|h|crop (optional)
+    $resizearg = false;
+
+    // Don't fire in wp-admin
+    if (!is_admin()) {
+
+        if ('core/image' !== $block['blockName']) {
+            return $block_content; // Only modify core/image blocks
+        }
+
+        // Extract the image ID from the wp-image-* class within the img tag
+        if (preg_match('/<img[^>]* class="[^"]*wp-image-(\d+)[^"]*"/', $block_content, $matches)) {
+            $image_id = $matches[1];
+        } else {
+            return $block_content; // No image ID found, return original content
+        }
+
+        // Extract the size class from the figure tag's class attribute
+        if (preg_match('/<figure[^>]* class="[^"]*size-([^\s"]+)[^"]*"/', $block_content, $size_matches)) {
+            $size_name = $size_matches[1];
+        } else {
+            return $block_content; // No size found, return original content
+        }
+
+        // Get the image metadata
+        $metadata = wp_get_attachment_metadata($image_id);
+        if (!$metadata || !isset($metadata['sizes'][$size_name])) {
+            return $block_content; // No metadata found for this size, return original content
+        }
+
+        // Get the width, height, and file from the metadata
+        $width = $metadata['sizes'][$size_name]['width'];
+        $height = $metadata['sizes'][$size_name]['height'];
+        $new_file = $metadata['sizes'][$size_name]['file'];
+
+        if ($resizearg) {
```

Comment:
This condition is always `false` because we don't seem to set `$resizearg` to `true` anywhere.

Reply:
Indeed, we don't set it true anywhere - yet. I'm deferring to y'all as to how we should let the user set this (if we include it at all).